src/utils/refreshDockerode.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DockerOptions } from 'dockerode';
+import Dockerode = require('dockerode');
+import { ext } from '../extensionVariables';
+import { addDockerSettingsToEnv } from './addDockerSettingsToEnv';
+import { exec } from './exec';
+import { isWindows } from './osUtils';
+
+const unix = 'unix://';
+const npipe = 'npipe://';
+
+const SSH_URL_REGEX = /ssh:\/\//i;
+
+// Not exhaustive--only the properties we're interested in
+interface IDockerEndpoint {
+    Host?: string;
+}
+
+// Also not exhaustive--only the properties we're interested in
+interface IDockerContext {
+    Endpoints: { [key: string]: IDockerEndpoint }
+}
+
+/**
+ * Dockerode parses and handles the well-known `DOCKER_*` environment variables, but it doesn't let us pass those values as-is to the constructor
+ * Thus we will temporarily update `process.env` and pass nothing to the constructor
+ */
+export async function refreshDockerode(): Promise<void> {
+    const oldEnv = process.env;
+    try {
+        process.env = { ...process.env }; // make a clone before we change anything
+        addDockerSettingsToEnv(process.env, oldEnv);
+        ext.dockerodeInitError = undefined;
+        ext.dockerode = new Dockerode(await getDockerodeOptions());

```

Comment:
Since this method is now async, is there a chance that the now modified `process.env` is exposed more broadly than desired?

Reply:
Good point. I'll look for a way to rearrange things so there's no async code inbetween the environment changes.

Reply:
I think it is fixed now, double check my work.